analyzer/src/main/kotlin/managers/GoMod.kt

```diff
@@ -297,12 +298,8 @@ private fun getRevision(version: String): String {
     return version
 }
 
-internal fun Identifier.toVcsInfo(): VcsInfo =
-    if (name.startsWith("github.com")) {
-        val projectUrl = "https://${name.removeSuffix("/")}"
-        val vcsInfo = VcsHost.toVcsInfo(projectUrl)
-
-        vcsInfo.copy(revision = getRevision(version))
-    } else {
-        VcsInfo.EMPTY
-    }
+internal fun Identifier.toVcsInfo(): VcsInfo {
+    val projectUrl = name.takeIf { it.startsWith("http") } ?: "https://$name"
```

Comment:
This code can now potentially be called for VCS types other than `git`, e.g. subversion.
IIUC the project URL construction for subversion does not work as the URL should start with 'svn:'?

Reply:
I've generalized the check accordingly.

Reply:
Maybe one step back. Your change seems to assume that the `name` part of the `Identifier` can have a `protocol` prefix.
Is that assumption actually reasonable to make?

Apart from that, do `https` URLs work for  SVN?

Reply:
> Your change seems to assume that the `name` part of the `Identifier` can have a `protocol` prefix. Is that assumption actually reasonable to make?

I don't know enough about Go to tell. I just didn't want prepend a protocol without checking whether it's already there.

*Edit:* But according to https://golang.org/ref/mod#go-mod-file-ident ("limited ASCII punctuation") it can never contain a protocol as ":" is not allowed. So I'll simply drop the check.

> Apart from that, do `https` URLs work for SVN?

[Yes](https://subversion.apache.org/quick-start#what-is-a-repository).

Reply:
Thanks!